### PR TITLE
Add support for concatenating single chars onto strings #484

### PR DESCRIPTION
**Pull request by c0rp3n**
Partially addresses #484 though this does not support concatenating enums as this is not handled in the lexer like the current string concatenation is.
It would seem the string concatenation would have to be moved to a later stage of compilation so the value of the enum is know for that.
But this pull request does support char literals so defines could be used for the usage in #484.

Also added a test for this functionality `tests/basic/cat-str-char.sp` I've inspected the output of this and they are correct and haven't seen anyother side effects.

### CODE REVIEW

**Comment by Fyren** [comment]
It would be helpful if you made the test check the output instead of being compile-only.  You just have to put the expected output in a file with the same name except with `out` as the extension.

**Comment by c0rp3n** [comment]
Added a out file to the test and moved it to the correct folder, also realised that I type out the wrong char code so that got fixed too.

File: compiler/lexer.cpp [hunk 1]
```diff
@@ -2187,24 +2187,38 @@ lex_string_literal(full_token_t* tok, cell* lexvalue)
...
 
         static char buffer[sLINEMAX + 1];
         char* cat = buffer;
-        while (*lptr != '\"' && *lptr != '\0' && (cat - buffer) < sLINEMAX) {
-            if (*lptr != '\a') { /* ignore '\a' (which was inserted at a line concatenation) */
-                *cat++ = *lptr;
-                if (*lptr == sc_ctrlchar && *(lptr + 1) != '\0')
-                    *cat++ = *++lptr; /* skip escape character plus the escaped character */
+        if (*lptr == '\"')
+        {
```
Comment (dvander) [inline compiler/lexer.cpp:2195]: Nit: brace on previous line:
` if (*lptr == '"') {`

File: compiler/lexer.cpp [hunk 2]
```diff
@@ -2187,24 +2187,38 @@ lex_string_literal(full_token_t* tok, cell* lexvalue)
...
+                    *cat++ = *lptr;
+                    if (*lptr == sc_ctrlchar && *(lptr + 1) != '\0')
+                        *cat++ = *++lptr; /* skip escape character plus the escaped character */
+                }
+                lptr++;
             }
-            lptr++;
+        }
+        else
+        {
```
Comment (dvander) [inline compiler/lexer.cpp:2207]: Nit, collapse into one line:
`} else {`

**Comment by c0rp3n** [comment]
Corrected the code styling / formatting to match the project.

**Comment by dvander** [comment]
Thanks, looks good!

**Comment by TheByKotik** [comment]
I just wanna reports it for future pr's.
```
	PrintToServer( '.'...'.' ); // Error.
	PrintToServer( '.'..."." ); // Error.
	PrintToServer( "."...'.' ); // Correct.
```

**Pull request merged**